utils/rpms/daos.spec

```diff
@@ -211,14 +211,22 @@ This is the package needed to build software with the DAOS library.
 %build
 
 %define conf_dir %{_sysconfdir}/daos
+%if (0%{?rhel} >= 7)
+%define scons_exe scons-3
+%else
+%define scons_exe scons
+%endif
+%{scons_exe} %{?_smp_mflags} \
+      --config=force         \
+      --no-rpath             \
+      CONF_DIR=%{conf_dir}   \
+      PREFIX=%{buildroot}    \
+     %{?scons_args}          \
+     %{?compiler_args}
 
-scons %{?_smp_mflags}      \
-      --config=force       \
-      --no-rpath           \
-      USE_INSTALLED=all    \
-      CONF_DIR=%{conf_dir} \
-      PREFIX=%{buildroot} \
-      %{?scons_args}
+%if ("%{?compiler_args}" == "COMPILER=covc")
+cp -f test.cov /tmp/
+%endif
```

Comment:
Why is this being copied here (and then again from `/tmp/test.cov` to `%{?buildroot}/usr/lib/daos/TESTING/ftest/`) on line 244?  Can the copy on line 244 not just be from the current directory instead of routing it through `/tmp/`?

Reply:
So both the %build and %install phases generate test.cov, and in this case the one from %install will overwrite the earlier. The one we want is the one we build tests with i.e. the one from %build, so copying it to a temp area.

Reply:
If the `%install` phase is not actually doing any rebuilding, as it should not be (see previous discussion) then why would that file be getting regenerated in the `%install` stage?  Once the `%build` stage's `scons` has built everything, the `%install` stage's `scons` should be doing nothing more than copying files to their installation location.  Since there should be no compiling actually happening in `%install`, there should be no generation of a new `test.cov` in `%install`.  Ultimately, if there is, then the `%install` stage is still trying to [re-]build, which it should not be doing.

Reply:
The install is not rebuilding daos. I have verified this in the build.log. We turn bullseye on in compiler setup in scons. As a result, test.cov is generated whenever we run scons with covc, whether just build or install. Maybe we need to revisit the setup in scons but I think that may involve changes in one or more scons scripts, and also more testing. 
To keep this PR to a smaller scope, is there a reason we cannot implement this approach of copying the build test.cov to /tmp first , in order to work with what we have in scons today?

Reply:
> The install is not rebuilding daos. I have verified this in the build.log.

When Bullseye is enabled?  What is the most recent run with that enabled, so that I can see what it's doing?

Reply:
Yea, I pushed a commit with bullseye enabled. You should be able to see the log in the next build.

Reply:
Looks like coverage build broke with a commit landed 2 days ago fcf4b80. Need to talk to Jeff when he gets back.

Reply:
Fix was landed this morning. PR merged with master and the build.log with cov turned on can be found at
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5021/10/artifact/artifacts/centos7/build.log/*view*/

Reply:
So yes, it looks like `%install` is actually just doing installation now.  Excellent.  Have you confirmed/verified that `%install` is actually still generating a `test.cov` file and that it's different than `%build`'s?

You could `ls -l` and `md5sum` `test.cov` at the end of both `%build` and `%install` to confirm that `%install` is actually generating a new and different one.  But frankly I cannot see how it could be since nothing is compiling in `%install` now.

I wonder if `scons` is running it in `%install` during it's _checking_ phase and creating a new `test.cov`.

In any case, using `/tmp` as a/the temporary holding space is probably unnecessary and slightly more prone to accident (it would race with another build happening on the same machine) than simply renaming it in the current workspace: `mv test.cov{,-build}` and then moving it from that name in `%install`.

Reply:
> I can push a fix to rename file, instead of putting it in /tmp

Excellent.

> I tested it in a separate PR and on 1.2 as well.

Ah, great.  Maybe that is good enough to look at.  Which PR?

Reply:
You can look at the 1.2 PR which contains the same changes. I started it to make sure nothing breaks there when we cherry pick this one after it lands.
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5208/7/
But I'm going to need to rerun this PR with tests anyway with the change in daos.spec to rename the test.cov file to make sure everything still works.

Reply:
Looks cool.  Can you determine from that output that code paths tested by functional testing are being reported and that that report isn't just what unit testing is testing?  If there is nothing in that report that would indicate that, I suppose you could do a run with bullseye enabled for functional testing and disabled for unit testing to see what coverage there is, couldn't you?

Reply:
The clover report in Jenkins right now if from unit testing. This PR is to enable coverage file generation for functional tests. There will be follow up PR to merge and publish the coverage files in Jenkins.

Reply:
> The clover report in Jenkins right now if from unit testing. This PR is to enable coverage file generation for functional tests. There will be follow up PR to merge and publish the coverage files in Jenkins.

So we don't know that this PR is actually generating and packaging files into the RPM that will actually generate the coverage data in Functional testing?  Is there any reason to land this without the "followup PR" rather than just combining them?

Reply:
We do know the coverage of functional testing from this PR. I can generate the report locally from the coverage collected in this PR. I prefer to keep smaller scope in PRs. This one is to enable the generation/collection of coverage. A separate one for publishing in Jenkins.

Reply:
Ahh.  OK.  If you've manually tested, that's fair enough.